### PR TITLE
feat: Add ADR for architectural plans

### PR DESCRIPTION
**Pull request by technoplato**
### TL;DR

This ADR introduces a new architectural approach for the group joining feature, inspired by the Composable Architecture pattern.

### What changed?

- Introduced ADR-002 to document the architectural refactoring for the group joining feature
- Proposed implementation of a state machine using XState for group joining logic
- Introduced a dependency control pattern for managing side effects and external dependencies
- Proposed development of an event-driven testing framework
- Outlined integration with existing Zustand and React Query logic
- Provided a detailed migration plan and implementation details

### Why make this change?

- Improve testability by separating business logic from UI
- Enhance maintainability through consistent patterns
- Increase scalability to handle future growth and complexity
- Provide better visibility for non-technical stakeholders through state machine diagrams

## Summary by CodeRabbit

- **New Features**
	- Introduced a proposal for refactoring the group joining feature to improve application architecture.
	- Implemented a state machine to manage the group joining process, enhancing user experience.
	- Developed an event-driven testing framework for better test coverage.

- **Bug Fixes**
	- Removed outdated functionality related to group invite management to streamline the application.

- **Documentation**
	- Updated documentation to reflect architectural changes and provided training for the team.

### CODE REVIEW

**Comment by technoplato** [comment]
* **#1068** <a href="https://app.graphite.dev/github/pr/ephemeraHQ/converse-app/1068?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @technoplato and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Review by technoplato — COMMENTED** [review]
Note: The only PR where it's vital that checks pass is the top most PR (#1073) in this case: https://app.graphite.dev/github/pr/ephemeraHQ/converse-app/1073

Some of the Types cleanup was missed, and it would be messy to move things around in this PR.

File: adr/adr.002.architecture for business logic and dependencies - join group/adr.002.architecture-poc-for-business-logic-and-dependencies-group-joining.md [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+## Date
+
+- **Proposed**: 2024-09-30
+- **Decided**: 2024-10-01
+- **Work Began**: 2024-09-26
+- **Work Completed**: [To be filled after completion]
+
+## Context
+
+During our team offsite in Nashville from **September 23rd to 27th, 2024**, we discussed opportunities to improve our application's architecture. Michael, a new team member, brings experience in implementing patterns that increase testability and separate business logic from UI, enhancing code robustness. He also introduced a pattern for controlling dependencies and side effects in applications.
```
Comment (technoplato) [inline adr/adr.002.architecture for business logic and dependencies - join group/adr.002.architecture-poc-for-business-logic-and-dependencies-group-joining.md:16]: demo comment: I don't like that you mentioned Nashville here. Please remove

**Comment by technoplato** [comment]
### Merge activity

* **Oct 31, 8:14 PM EDT**: [Graphite](https://app.graphite.dev/github/pr/ephemeraHQ/converse-app/1068) rebased this pull request as part of a merge.

**Comment by alexrisch** [comment]
Duplicate

**Pull request closed without merging**